File: mpp-proiect-csharp-culbec/Server/Service/ServiceUserImpl.cs
using Common.Exceptions;
using log4net;
using Model;
using Persistence.Repository;

namespace Server.Service;

public class ServiceUserImpl : IServiceUser
{
    private static readonly ILog Log = LogManager.GetLogger(typeof(ServiceUserImpl));
    private readonly IRepositoryUser _repositoryUser;

    public ServiceUserImpl(IRepositoryUser repositoryUser)
    {
        Log.Info("Initializing ServiceUserImpl.");
        _repositoryUser = repositoryUser;
        Log.Info("ServiceUserImpl initialized.");
    }

    public User Login(string username, string password)
    {
        Log.InfoFormat("Attempting to login user with username {0}.", username);

        try
        {
            var user = _repositoryUser.FindUserByCredentials(username, password);

            if (user == null)
            {
                Log.Error("The user couldn't be found.");
                throw new ServiceException("The user couldn't be found.");
            }

            Log.Info("User logged in successfully.");
            return user;
        }
        catch (RepositoryException e)
        {
            Log.ErrorFormat("Something went wrong when trying to login user with username {0}: {1}", username,
                e.Message);
            throw new ServiceException("Something went wrong when trying to login user with username " + username +
                                       ": " + e.Message);
        }
    }
}

File: mpp-proiect-java-culbec/MotorcycleContest/Server/src/main/java/Service/ServiceUserImpl.java
package Service;

import Repository.RepositoryException;
import Exceptions.ServiceException;
import Model.User;
import Repository.UserRepository;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.util.Optional;

public class ServiceUserImpl implements ServiceUser {
    private static final Logger LOGGER = LogManager.getLogger();
    private final UserRepository userRepository;

    public ServiceUserImpl(UserRepository userRepository) {
        LOGGER.traceEntry("Initializing the ServiceUserImpl with the UserRepository...");
        this.userRepository = userRepository;
        LOGGER.traceExit("Initialized the ServiceUserImpl with the UserRepository!");
    }

    @Override
    public User login(String username, String password) throws ServiceException {
        LOGGER.traceEntry("Logging in the user with the username {}...", username);
        try {
            Optional<User> user = userRepository.findUserByCredentials(username, password);

            if (user.isEmpty()) {
                LOGGER.error("The user with the username {} doesn't exist or the password is incorrect!", username);
                throw new ServiceException("The user doesn't exist or the password is incorrect!");
            }

            LOGGER.traceExit("Logged in the user with the username {}!", username);
            return user.get();
        } catch (RepositoryException e) {
            LOGGER.error("Couldn't login the user with the username {}: {}", username, e.getMessage());
            throw new ServiceException(e.getMessage());
        }
    }
}
